lib/internal/util/comparisons.js

```diff
@@ -533,157 +525,267 @@ function mapMightHaveLoosePrim(a, b, prim, item2, memo) {
   return !b.has(altValue) && innerDeepEqual(item1, item2, kLoose, memo);
 }
 
-function partialObjectSetEquiv(a, b, mode, set, memo) {
+function partialObjectSetEquiv(array, a, b, mode, memo) {
   let aPos = 0;
-  for (const val of a) {
+  let direction = 1;
+  let start = 0;
+  let end = array.length - 1;
+  for (const val1 of a) {
     aPos++;
-    if (!b.has(val) && setHasEqualElement(set, val, mode, memo) && set.size === 0) {
-      return true;
+    if (!b.has(val1)) {
+      let innerStart = start;
+      if (direction === 1) {
+        if (innerDeepEqual(val1, array[start], mode, memo)) {
+          if (end === start) {
+            return true;
+          }
+          start += 1;
+          continue;
+        }
+        if (start === end) {
+          continue;
```

Comment:
Shouldn't this be a return false?

Reply:
I had another look. It is correct as is. It is about matching the last entry of set b with a later element of set a.